cpp/src/arrow/json/reader_test.cc

```diff
@@ -274,5 +275,37 @@ TEST(ReaderTest, ListArrayWithFewValues) {
   AssertTablesEqual(*actual_table, *expected_table);
 }
 
+TEST_P(ReaderTest, UnquotedDecimal) {
+  parse_options_.unexpected_field_behavior = UnexpectedFieldBehavior::InferType;
```

Comment:
I don't think this line is useful since we're passing the entire expected schema?

Reply:
Removed. Also removed the one for MixedDecimal.